torch/testing/_asserts.py

```diff
@@ -187,6 +294,10 @@ def _equalize_attributes(actual: Tensor, expected: Tensor) -> Tuple[Tensor, Tens
         actual = actual.to(dtype)
         expected = expected.to(dtype)
 
+    if actual.is_sparse and actual.is_coalesced() != expected.is_coalesced():
```

Comment:
Shouldn't the `expected` content drive the testing logic? I suggest:
```
if expected.is_sparse and ...
```

Reply:
That doesn't matter here, because after `_check_attributes_equal()` `actual.is_sparse == expected.is_sparse`. Otherwise we would have bailed out there.

Reply:
yeah, the suggestion would be just a nit..